python/core/qgsstringstatisticalsummary.sip

```diff
@@ -1,165 +1,220 @@
- *
- * \note Added in version 2.16
- */
+/************************************************************************
+ * This file has been generated automatically from                      *
+ *                                                                      *
+ * src/core/qgsstringstatisticalsummary.h                               *
+ *                                                                      *
+ * Do not edit manually ! Edit header and run scripts/sipify.pl again   *
+ ************************************************************************/
+
+
+
+
 
 
 class QgsStringStatisticalSummary
 {
-%TypeHeaderCode
-#include <qgsstringstatisticalsummary.h>
+%Docstring
+ Calculator for summary statistics and aggregates for a list of strings.
+
+ Statistics are calculated by calling calculate() and passing a list of strings. The
+ individual statistics can then be retrieved using the associated methods. Note that not all statistics
+ are calculated by default. Statistics which require slower computations are only calculated by
+ specifying the statistic in the constructor or via setStatistics().
+
+.. versionadded:: 2.16
 %End
 
+%TypeHeaderCode
+#include "qgsstringstatisticalsummary.h"
+%End
   public:
 
-    public:
-
-    //! Enumeration of flags that specify statistics to be calculated
     enum Statistic
     {
-      Count,  //!< Count
-      CountDistinct,  //!< Number of distinct string values
-      CountMissing,  //!< Number of missing (null) values
-      Min, //!< Minimum string value
-      Max, //!< Maximum string value
-      MinimumLength, //!< Minimum length of string
-      MaximumLength, //!< Maximum length of string
-      MeanLength, //!< Mean length of strings
-      All, //! All statistics
+      Count,
+      CountDistinct,
+      CountMissing,
+      Min,
+      Max,
+      MinimumLength,
+      MaximumLength,
+      MeanLength,
+      All,
     };
     typedef QFlags<QgsStringStatisticalSummary::Statistic> Statistics;
 
-    /** Constructor for QgsStringStatistics
-     * @param stats flags for statistics to calculate
-     */
+
     QgsStringStatisticalSummary( QgsStringStatisticalSummary::Statistics stats = All );
+%Docstring
+ Constructor for QgsStringStatistics
+ \param stats flags for statistics to calculate
+%End
 
-    /** Returns flags which specify which statistics will be calculated. Some statistics
-     * are always calculated (e.g., count).
-     * @see setStatistics
-     */
     Statistics statistics() const;
+%Docstring
+ Returns flags which specify which statistics will be calculated. Some statistics
+ are always calculated (e.g., count).
+.. seealso:: setStatistics
+ :rtype: Statistics
+%End
 
-    /** Sets flags which specify which statistics will be calculated. Some statistics
-     * are always calculated (e.g., count).
-     * @param stats flags for statistics to calculate
-     * @see statistics
-     */
     void setStatistics( Statistics stats );
```

Comment:
you might need to add `QgsStatisticalSummary`in front of `Statistics` (in the header)